src/index.d.ts

```diff
@@ -17,26 +17,26 @@ declare module 'marked-react' {
     elementId: number;
 
     constructor(options?: MarkdownOptions);
-    heading(children: any, level: any): ReactElement;
-    paragraph(children: any): ReactElement;
-    link(href: any, text: any): ReactElement;
-    image(href: any, text: any, title: any): ReactElement;
-    codespan(code: any, lang: any): ReactElement;
-    code(code: any, lang: any): ReactElement;
-    blockquote(children: any): ReactElement;
-    list(children: any, ordered: any): ReactElement;
-    listItem(children: any): ReactElement;
-    checkbox(checked: any): ReactElement;
-    table(children: any): ReactElement;
-    tableHeader(children: any): ReactElement;
-    tableBody(children: any): ReactElement;
-    tableRow(children: any): ReactElement;
-    tableCell(children: any, flags: any): ReactElement;
-    strong(children: any): ReactElement;
-    em(children: any): ReactElement;
-    del(children: any): ReactElement;
-    text(text: any): any;
-    html(html: any): any;
+    heading(children: ReactNode[], level: 1 | 2 | 3 | 4 | 5 | 6): ReactElement;
```

Comment:
A question here. Should this be an array or just `ReactNode`?

Reply:
`children` is the third argument of the [function](https://github.com/sibiraj-s/marked-react/blob/master/src/ReactRenderer.js#L34) `createElement()`.
Function `createElement()` has [type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L254-L257) `ReactNode[]` for `...children` argument.

Reply:
> Function createElement() has [type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L254-L257) ReactNode[] for ...children argument.

Yes. createElement can accept more than 3 arguments hence the children is spreaded and has array type

but in our case we are not spreading the children createElement, it's just the 3rd argument. I guess it should be just `ReactNode`.

Reply:
Let me check this out. Not a big TS user.

Reply:
<img width="622" alt="Screenshot 2022-08-17 at 1 32 00 PM" src="https://user-images.githubusercontent.com/20282546/185067161-e0b96d11-fcee-4de7-ae42-fe08b1e8bac5.png">

Just got clarified. The reason for that is `createElement` can have more than one children. but in our case it is going to be always the 3rd argument. so it should `ReactNode` should do IMO. thoughts?

Reply:
Yes, we can get only one `ReactNode`.
So, every function of `ReactRenderer` takes one `ReactNode`? `list` for example.

Reply:
ReactNode will also work for an array of children, because ReactNode can be of type ReactFragment, which is nothing but `Iterable<ReactNode>`. 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/eee1c75d89e90fc6b5a5112c2bb8b4b5197c3bca/types/react/index.d.ts#L229-L230

If we need to be explicit we can have both `ReactNode | ReactNode[]` for list, tableRow, tableHeader,  tableBody ad the rest can be just `ReactNode`.